src/main/java/screens/WeekViewScreen.java

```diff
@@ -0,0 +1,304 @@
+
+     */
+
+    private void addButtons(JToolBar toolBar) {
+        JButton newTask = new JButton("New Task");
+        newTask.addActionListener(this);
+        newTask.setActionCommand("create task");
+        toolBar.add(newTask);
+
+        JButton newEvent = new JButton("New Event");
+        newEvent.addActionListener(this);
+        newEvent.setActionCommand("create event");
+        toolBar.add(newEvent);
+
+        JButton clear = new JButton("Clear Week");
+        clear.addActionListener(this);
+        clear.setActionCommand("clear all");
+        toolBar.add(clear);
+    }
+
+
+    /**
+     * Invoked when an action occurs.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getActionCommand().equals("create event")) {
+            screens.get("create event").setVisible(true);
+        } else if (e.getActionCommand().equals("create task")) {
+            screens.get("create task").setVisible(true);
+        } else if (e.getActionCommand().equals("clear all")) {
+            int confirmed = JOptionPane.showConfirmDialog(null,
+                    "Are you sure you want to clear this week?", "Clear Week Message Box",
+                    JOptionPane.YES_NO_OPTION);
+
+            if (confirmed == JOptionPane.YES_OPTION) {
+                ((ClearAllController)controllers.get("clear all")).clearAll();
+                JOptionPane.showMessageDialog(new JFrame(), "Week cleared");
+            }
+        } else if (e.getActionCommand().contains("event")) {
+
+            // the following should be later changed to get all these values from the view model but at present they are
+            // encoded in a string that is saved as the action command for a task or event.
+            ((ClickEventScreen)screens.get("click event")).setDayIndex(e.getActionCommand().charAt(6) - 48);
+            ((ClickEventScreen)screens.get("click event")).setStartTime(e.getActionCommand().substring(8,13));
+            ((ClickEventScreen)screens.get("click event")).setEndTime(e.getActionCommand().substring(14,19));
+            ((ClickEventScreen)screens.get("click event")).setEventTitle(e.getActionCommand().substring(20));
+
+            screens.get("click event").setVisible(true);
+        } else if (e.getActionCommand().contains("task")) {
+            ((ClickTaskScreen)screens.get("click task")).setDayIndex(e.getActionCommand().charAt(5) - 48);
+            ((ClickTaskScreen)screens.get("click task")).setTaskCompleted(
+                    Boolean.parseBoolean(e.getActionCommand().substring(7, 8)));
+            ((ClickTaskScreen)screens.get("click task")).setTaskTitle(e.getActionCommand().substring(9));
+
+            screens.get("click task").setVisible(true);
+        }
+    }
+
+    /**
+     * Invoked the first time a window is made visible.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void windowOpened(WindowEvent e) {
```

Comment:
Is this method required? The body is empty.

Reply:
Because WeekViewScreen implements WindowListener, I have to implement all its methods. Since I only want special things to happen when the user closes the window, not when they open or minimize, etc. the window, I left the implementations of the others empty since they still need to be in the class.

Reply:
I suppose I could make it an abstract class, but I'm not sure that's the best option.

Reply:
Perhaps leave the empty methods for the time being. Turning it into an abstract would merit raising another issue on GitHub.